--- origsrc/SDL2-2.0.3/configure.in	2015-06-11 21:49:52.209555000 -0500
+++ src/SDL2-2.0.3/configure.in	2015-06-11 21:50:07.969556300 -0500
@@ -79,29 +79,6 @@ elif test -d .hg; then
 fi
 BASE_CFLAGS=""
 BASE_LDFLAGS=""
-case "$host" in
-    *-*-cygwin*)
-        # We build SDL on cygwin without the UNIX emulation layer
-        save_CFLAGS="$CFLAGS"
-        have_no_cygwin=no
-        AC_MSG_CHECKING(for GCC -mno-cygwin option)
-        CFLAGS="$save_CFLAGS -mno-cygwin"
-
-        AC_TRY_COMPILE([
-        ],[
-        ],[
-        have_no_cygwin=yes
-        ])
-        AC_MSG_RESULT($have_no_cygwin)
-        CFLAGS="$save_CFLAGS"
-
-        if test x$have_no_cygwin = xyes; then
-            BASE_CFLAGS="-mno-cygwin"
-            BASE_LDFLAGS="-mno-cygwin"
-        fi
-        BASE_CFLAGS="$BASE_CFLAGS -I/usr/include/mingw"
-        ;;
-esac
 # Uncomment the following line if you want to force SDL and applications
 # built with it to be compiled for a particular architecture.
 #AX_GCC_ARCHFLAG([no], [BASE_CFLAGS="$BASE_CFLAGS $ax_cv_gcc_archflag]")
@@ -244,7 +221,7 @@ if test x$enable_libc = xyes; then
 
     dnl Checks for library functions.
     case "$host" in
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ;;
     *)
         AC_FUNC_ALLOCA
@@ -2624,7 +2601,7 @@ AC_HELP_STRING([--enable-rpath], [use an
 
 dnl Set up the configuration based on the host platform!
 case "$host" in
-    *-*-linux*|*-*-uclinux*|*-*-gnu*|*-*-k*bsd*-gnu|*-*-bsdi*|*-*-freebsd*|*-*-dragonfly*|*-*-netbsd*|*-*-openbsd*|*-*-sysv5*|*-*-solaris*|*-*-hpux*|*-*-aix*|*-*-minix*)
+    *-*-linux*|*-*-uclinux*|*-*-gnu*|*-*-k*bsd*-gnu|*-*-bsdi*|*-*-freebsd*|*-*-dragonfly*|*-*-netbsd*|*-*-openbsd*|*-*-sysv5*|*-*-solaris*|*-*-hpux*|*-*-aix*|*-*-minix*|*-*-cygwin*)
         case "$host" in
             *-raspberry-linux*)
                 # Raspberry Pi
@@ -2656,6 +2633,7 @@ case "$host" in
             *-*-hpux*)          ARCH=hpux ;;
             *-*-aix*)           ARCH=aix ;;
             *-*-minix*)         ARCH=minix ;;
+            *-*-cygwin*)        ARCH=cygwin ;;
         esac
         CheckVisibilityHidden
         CheckDummyVideo
@@ -2759,7 +2737,7 @@ case "$host" in
             SOURCES="$SOURCES $srcdir/src/core/linux/SDL_evdev.c"
         fi       
         ;;
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-mingw32*)
         ARCH=win32
         if test "$build" != "$host"; then # cross-compiling
             # Default cross-compile location
--- origsrc/SDL2-2.0.3/include/SDL_platform.h	2014-03-15 21:31:42.000000000 -0500
+++ src/SDL2-2.0.3/include/SDL_platform.h	2015-06-11 21:50:07.976057100 -0500
@@ -113,8 +113,12 @@
 #undef __SOLARIS__
 #define __SOLARIS__ 1
 #endif
+#if defined(__CYGWIN__)
+#undef __CYGWIN__
+#define __CYGWIN__ 1
+#endif
 
-#if defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__)
+#if defined(WIN32) || defined(_WIN32)
 /* Try to find out if we're compiling for WinRT or non-WinRT */
 /* If _USING_V110_SDK71_ is defined it means we are using the v110_xp or v120_xp toolset. */
 #if defined(__MINGW32__) || (defined(_MSC_VER) && (_MSC_VER >= 1700) && !_USING_V110_SDK71_)	/* _MSC_VER==1700 for MSVC 2012 */
@@ -131,7 +135,7 @@
 #undef __WINDOWS__
 #define __WINDOWS__   1
 #endif /* _MSC_VER < 1700 */
-#endif /* defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__) */
+#endif /* defined(WIN32) || defined(_WIN32) */
 
 #if defined(__WINDOWS__)
 #undef __WIN32__
--- origsrc/SDL2-2.0.3/src/dynapi/SDL_dynapi.c	2014-03-15 21:31:44.000000000 -0500
+++ src/SDL2-2.0.3/src/dynapi/SDL_dynapi.c	2015-06-12 00:19:47.269282500 -0500
@@ -198,7 +198,7 @@ SDL_DYNAPI_entry(Uint32 apiver, void *ta
 
 /* Obviously we can't use SDL_LoadObject() to load SDL.  :)  */
 /* Also obviously, we never close the loaded library. */
-#if defined(WIN32) || defined(_WIN32) || defined(__CYGWIN__)
+#if defined(WIN32) || defined(_WIN32)
 #ifndef WIN32_LEAN_AND_MEAN
 #define WIN32_LEAN_AND_MEAN 1
 #endif
--- origsrc/SDL2-2.0.3/test/configure.in	2014-03-15 21:31:44.000000000 -0500
+++ src/SDL2-2.0.3/test/configure.in	2015-06-11 21:50:07.980557700 -0500
@@ -22,7 +22,13 @@ ISMACOSX="false"
 
 dnl Figure out which math library to use
 case "$host" in
-    *-*-cygwin* | *-*-mingw32*)
+    *-*-cygwin*)
+        ISUNIX="true"
+        EXE=".exe"
+        MATHLIB=""
+        SYS_GL_LIBS="-lGL"
+        ;;
+    *-*-mingw32*)
         ISWINDOWS="true"
         EXE=".exe"
         MATHLIB=""
