--- origsrc/SDL2-2.0.3/configure.in	2015-06-11 21:50:54.491963900 -0500
+++ src/SDL2-2.0.3/configure.in	2015-06-11 21:51:05.212825200 -0500
@@ -791,7 +791,7 @@ AC_HELP_STRING([--enable-esd], [support
             AC_ARG_ENABLE(esd-shared,
 AC_HELP_STRING([--enable-esd-shared], [dynamically load ESD audio support [[default=yes]]]),
                           , enable_esd_shared=yes)
-            esd_lib=[`find_lib "libesd.so.*" "$ESD_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            esd_lib=cygesd-0.dll
 
             AC_DEFINE(SDL_AUDIO_DRIVER_ESD, 1, [ ])
             SOURCES="$SOURCES $srcdir/src/audio/esd/*.c"
@@ -840,7 +840,7 @@ AC_HELP_STRING([--enable-pulseaudio], [u
             AC_ARG_ENABLE(pulseaudio-shared,
 AC_HELP_STRING([--enable-pulseaudio-shared], [dynamically load PulseAudio support [[default=yes]]]),
                           , enable_pulseaudio_shared=yes)
-            pulseaudio_lib=[`find_lib "libpulse-simple.so.*" "$PULSEAUDIO_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            pulseaudio_lib=cygpulse-0.dll
 
             AC_DEFINE(SDL_AUDIO_DRIVER_PULSEAUDIO, 1, [ ])
             SOURCES="$SOURCES $srcdir/src/audio/pulseaudio/*.c"
@@ -851,7 +851,7 @@ AC_HELP_STRING([--enable-pulseaudio-shar
             fi
             if test x$have_loadso = xyes && \
                test x$enable_pulseaudio_shared = xyes && test x$pulseaudio_lib != x; then
-                echo "-- dynamic libpulse-simple -> $pulseaudio_lib"
+                echo "-- dynamic libpulse -> $pulseaudio_lib"
                 AC_DEFINE_UNQUOTED(SDL_AUDIO_DRIVER_PULSEAUDIO_DYNAMIC, "$pulseaudio_lib", [ ])
                 SUMMARY_audio="${SUMMARY_audio} pulse(dynamic)"
 
@@ -954,7 +954,7 @@ AC_HELP_STRING([--enable-nas], [support
             AC_ARG_ENABLE(nas-shared,
 AC_HELP_STRING([--enable-nas-shared], [dynamically load NAS audio support [[default=yes]]]),
                           , enable_nas_shared=yes)
-            nas_lib=[`find_lib "libaudio.so.*" "$NAS_LIBS" | sed 's/.*\/\(.*\)/\1/; q'`]
+            nas_lib=cygaudio-2.dll
 
             if test x$have_loadso != xyes && \
                test x$enable_nas_shared = xyes; then
@@ -1333,6 +1333,16 @@ AC_HELP_STRING([--enable-x11-shared], [d
                     xss_lib='libXss.so'
                     xvidmode_lib='libXxf86vm.so'
                     ;;
+                *-*-cygwin*)
+                    x11_lib=cygX11-6.dll
+                    x11ext_lib=cygXext-6.dll
+                    xcursor_lib=cygXcursor-1.dll
+                    xinerama_lib=cygXinerama-1.dll
+                    xinput_lib=cygXi-6.dll
+                    xrandr_lib=cygXrandr-2.dll
+                    xrender_lib=cygXrender-1.dll
+                    xss_lib=cygXss-1.dll
+                    ;;
                 *)
                     x11_lib=[`find_lib "libX11.so.*" "$X_LIBS -L/usr/X11/$base_libdir -L/usr/X11R6/$base_libdir" | sed 's/.*\/\(.*\)/\1/; q'`]
                     x11ext_lib=[`find_lib "libXext.so.*" "$X_LIBS -L/usr/X11/$base_libdir -L/usr/X11R6/$base_libdir" | sed 's/.*\/\(.*\)/\1/; q'`]
@@ -2667,6 +2677,12 @@ case "$host" in
         # Set up files for the audio library
         if test x$enable_audio = xyes; then
           case $ARCH in
+            cygwin)
+                AC_DEFINE(SDL_AUDIO_DRIVER_WINMM, 1, [ ])
+                SOURCES="$SOURCES $srcdir/src/audio/winmm/*.c"
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lwinmm"
+                have_audio=yes
+            ;;
             sysv5|solaris|hpux)
                 AC_DEFINE(SDL_AUDIO_DRIVER_SUNAUDIO, 1, [ ])
                 SOURCES="$SOURCES $srcdir/src/audio/sun/*.c"
@@ -2687,6 +2703,12 @@ case "$host" in
         # Set up files for the joystick library
         if test x$enable_joystick = xyes; then
           case $ARCH in
+            cygwin)
+                AC_DEFINE(SDL_JOYSTICK_WINMM, 1, [ ])
+                SOURCES="$SOURCES $srcdir/src/joystick/windows/SDL_mmjoystick.c"
+                EXTRA_LDFLAGS="$EXTRA_LDFLAGS -lwinmm"
+                have_joystick=yes
+            ;;
             linux)
                 AC_DEFINE(SDL_JOYSTICK_LINUX, 1, [ ])
                 SOURCES="$SOURCES $srcdir/src/joystick/linux/*.c"
--- origsrc/SDL2-2.0.3/src/video/SDL_egl.c	2014-03-15 21:31:44.000000000 -0500
+++ src/SDL2-2.0.3/src/video/SDL_egl.c	2015-06-12 00:32:16.812962500 -0500
@@ -48,6 +48,14 @@
 #define DEFAULT_OGL_ES_PVR "libGLES_CM.dll"
 #define DEFAULT_OGL_ES "libGLESv1_CM.dll"
 
+#elif defined(__CYGWIN__)
+/* Cygwin */
+#define DEFAULT_OGL "cygGL-1.dll"
+#define DEFAULT_EGL "cygEGL-1.dll"
+#define DEFAULT_OGL_ES2 "cygGLESv2-2.dll"
+#define DEFAULT_OGL_ES_PVR "cygGLES_CM-1.dll"
+#define DEFAULT_OGL_ES "cygGLESv1_CM-1.dll"
+
 #else
 /* Desktop Linux */
 #define DEFAULT_OGL "libGL.so.1"
--- origsrc/SDL2-2.0.3/src/video/x11/SDL_x11opengl.c	2014-03-15 21:31:44.000000000 -0500
+++ src/SDL2-2.0.3/src/video/x11/SDL_x11opengl.c	2015-06-11 21:51:05.230827500 -0500
@@ -38,6 +38,8 @@
 #define DEFAULT_OPENGL  "/usr/X11R6/lib/libGL.1.dylib"
 #elif defined(__QNXNTO__)
 #define DEFAULT_OPENGL  "libGL.so.3"
+#elif defined(__CYGWIN__)
+#define DEFAULT_OPENGL  "cygGL-1.dll"
 #else
 #define DEFAULT_OPENGL  "libGL.so.1"
 #endif
--- origsrc/SDL2-2.0.5/src/core/linux/SDL_dbus.c	2014-03-15 21:31:44.000000000 -0500
+++ src/SDL2-2.0.5/src/core/linux/SDL_dbus.c	2015-06-11 21:51:05.235828200 -0500
@@ -41,9 +41,14 @@
 #include "SDL_dbus.h"
 
 #if SDL_USE_LIBDBUS
+#ifdef __CYGWIN__
+#define LIBDBUS_SONAME  "cygdbus-1-3.dll"
+#else
+#define LIBDBUS_SONAME  "libdbus-1.so.3"
+#endif
 /* we never link directly to libdbus. */
 #include "SDL_loadso.h"
-static const char *dbus_library = "libdbus-1.so.3";
+static const char *dbus_library = LIBDBUS_SONAME;
 static void *dbus_handle = NULL;
 static unsigned int screensaver_cookie = 0;
 static SDL_DBusContext dbus;
